.github/workflows/mobilecoin-dev-cd.yaml

```
@@ -0,0 +1,563 @@
+        CONSENSUS_ENCLAVE_PRIVKEY: ${{ github.workspace }}/.tmp/enclave_signing.pem
+        LEDGER_ENCLAVE_PRIVKEY: ${{ github.workspace }}/.tmp/enclave_signing.pem
+        VIEW_ENCLAVE_PRIVKEY: ${{ github.workspace }}/.tmp/enclave_signing.pem
+        INGEST_ENCLAVE_PRIVKEY: ${{ github.workspace }}/.tmp/enclave_signing.pem
+      run: |
+        cargo build --release \
+          -p mc-consensus-service \
+          -p mc-admin-http-gateway \
+          -p mc-util-generate-sample-ledger \
+          -p mc-util-grpc-admin-tool \
+          -p mc-util-keyfile \
+          -p mc-ledger-distribution \
+          -p mc-ledger-migration \
+          -p mc-ledger-from-archive \
+          -p mc-mobilecoind \
+          -p mc-watcher \
+          -p mc-fog-ingest-server \
+          -p mc-fog-ingest-client \
+          -p mc-fog-view-server \
+          -p mc-fog-report-server \
+          -p mc-fog-sql-recovery-db \
+          -p mc-fog-ledger-server \
+          -p mc-fog-test-client \
+          -p mc-fog-distribution \
+          -p mc-util-grpc-token-generator
+
+    - name: Copy artifacts to cache
+      if: |
+        steps.artifact_cache.outputs.cache-hit != 'true' &&
+        ! contains(github.event.head_commit.message, '[skip build]')
+      run: |
+        mkdir -p build_artifacts
+        find target/release -maxdepth 1 -executable -type f -exec cp "{}" build_artifacts/ \;
+        find target/release -maxdepth 1 -name "*.signed.so" -exec cp "{}" build_artifacts/ \;
+
+    - name: Create css measurements
+      if: |
+        steps.artifact_cache.outputs.cache-hit != 'true' &&
+        ! contains(github.event.head_commit.message, '[skip build]')
+      shell: bash
+      run: |
+        cd build_artifacts
+        for i in *.signed.so
+        do
+          css=$(echo -n "${i}" | sed -r 's/(.*)\.signed\.so/\1/')
+          sgx_sign dump -enclave "${i}" -dumpfile /dev/null -cssfile ${css}.css
+        done
+
+    - name: check artifacts
+      if: "! contains(github.event.head_commit.message, '[skip build]')"
+      run: |
+        ls -alR build_artifacts
+
+    - name: Upload Artifacts
+      if: "! contains(github.event.head_commit.message, '[skip build]')"
+      uses: actions/upload-artifact@v3
+      with:
+        name: rust-binaries
+        path: build_artifacts/
+
+  build-go-projects:
+    runs-on: [self-hosted, Linux, large]
+    container:
+      image: golang:1.16.4
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Add protobuf-compiler
+      if: "! contains(github.event.head_commit.message, '[skip build]')"
+      run: |
+        apt update
+        apt install -y protobuf-compiler zstd
+
+    - name: Cache go build binaries
+      if: "! contains(github.event.head_commit.message, '[skip build]')"
+      id: go_artifact_cache
+      uses: actions/cache@v3
+      with:
+        path: |
+          go_build_artifacts
+        # Key is a hash of all the .rs, .proto and Cargo.toml files.
```

nit: copy pasted comment from the rust version
```suggestion
        # Key is a hash of all the .go, go.*, and .proto files.
```